crates/solver/src/settlement_ranker.rs

```diff
@@ -60,6 +60,12 @@ impl SettlementRanker {
         let name = solver.name();
         match settlements {
             Ok(settlements) => {
+                if settlements.is_empty() {
+                    solver.notify_auction_result(
+                        auction_id,
+                        AuctionResult::Rejected(SolverRejectionReason::EmptySolution),
+                    );
+                }
```

Comment:
I did it here because this module is the only place where `notify_auction_result` was being called. So it felt more consistent. I think both approaches are terrible, and this function should be called based on a return value (that way you can't forget it, like we have).

Reply:
That function already gets called in that [file](https://github.com/cowprotocol/services/blob/main/crates/solver/src/solver/http_solver.rs#L145).
> I think both approaches are terrible

Yeah, definitely not great code. Fortunately this goes away with the new driver. 🤞